linera-core/src/client.rs

```diff
@@ -866,6 +872,16 @@ where
         for entry in response.info.requested_received_log {
             let query = ChainInfoQuery::new(entry.chain_id)
                 .with_sent_certificates_in_range(BlockHeightRange::single(entry.height));
+            let local_response = node_client
+                .handle_chain_info_query(query.clone())
+                .await
+                .map_err(|error| NodeError::LocalNodeQuery {
```

Comment:
There's already a conversion rule from `LocalNodeError` to `ChainClientError`.

Reply:
The problem is that we use this inside `communicate_with_quorum`, which requires us to return a `NodeError`, not `ChainClientError`. I guess I could make `communicate_with_quorum` generic in the error type, too?

Reply:
…but then `communicate_with_quorum` returns a `CommunicationError<E>`…

Reply:
Ok I just looked into generalizing `communicate_with_quorum`: this would require `ChainClientError` (hence `LocalNodeError`) to implement `Eq` and `Hash` which is annoying, so we can't do that. I think your approach is the most reasonable.